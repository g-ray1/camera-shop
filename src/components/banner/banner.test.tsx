import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Banner from './banner';

const product = {
  id: 1,
  name: 'Ретрокамера Dus Auge lV',
  previewImg: 'img/content/das-auge.jpg',
  previewImg2x: 'img/content/das-auge@2x.jpg',
  previewImgWebp: 'img/content/das-auge.webp',
  previewImgWebp2x: 'img/content/das-auge@2x.webp',
};

const description = 'Немецкий концерн BRW разработал видеокамеру Das Auge IV в начале 80-х годов, однако она до сих пор пользуется популярностью среди коллекционеров и яростных почитателей старинной техники. Вы тоже можете прикоснуться к волшебству аналоговой съёмки, заказав этот чудо-аппарат. Кто знает, может с Das Auge IV начнётся ваш путь к наградам всех престижных кинофестивалей.';

describe('Banner component', () => {
  it('Snapshot', () => {
    const view = render(<Banner product={product} description={description} />, {wrapper: BrowserRouter});

    expect(view).toMatchSnapshot();
  });

  it('Should return empty', () => {
    const view = render(<Banner />);

    expect(view.container).toBeEmptyDOMElement();
  });
});

