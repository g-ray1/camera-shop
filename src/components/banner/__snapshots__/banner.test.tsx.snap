// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="banner"
      >
        <picture>
          <source
            srcset="/img/content/das-auge.webp, /img/content/das-auge@2x.webp 2x"
            type="image/webp"
          />
          <img
            alt="баннер"
            height="280"
            src="img/content/das-auge.jpg"
            srcset="/img/content/das-auge@2x.jpg 2x"
            width="1280"
          />
        </picture>
        <p
          class="banner__info"
        >
          <span
            class="banner__message"
          >
            Новинка!
          </span>
          <span
            class="title title--h1"
          >
            Ретрокамера Dus Auge lV
          </span>
          <span
            class="banner__text"
          >
            Немецкий концерн BRW разработал видеокамеру Das Auge IV в начале 80-х годов, однако она до сих пор пользуется популярностью среди коллекционеров и яростных почитателей старинной техники. Вы тоже можете прикоснуться к волшебству аналоговой съёмки, заказав этот чудо-аппарат. Кто знает, может с Das Auge IV начнётся ваш путь к наградам всех престижных кинофестивалей.
          </span>
          <a
            class="btn"
            href="/product/1"
          >
            Подробнее
          </a>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="banner"
    >
      <picture>
        <source
          srcset="/img/content/das-auge.webp, /img/content/das-auge@2x.webp 2x"
          type="image/webp"
        />
        <img
          alt="баннер"
          height="280"
          src="img/content/das-auge.jpg"
          srcset="/img/content/das-auge@2x.jpg 2x"
          width="1280"
        />
      </picture>
      <p
        class="banner__info"
      >
        <span
          class="banner__message"
        >
          Новинка!
        </span>
        <span
          class="title title--h1"
        >
          Ретрокамера Dus Auge lV
        </span>
        <span
          class="banner__text"
        >
          Немецкий концерн BRW разработал видеокамеру Das Auge IV в начале 80-х годов, однако она до сих пор пользуется популярностью среди коллекционеров и яростных почитателей старинной техники. Вы тоже можете прикоснуться к волшебству аналоговой съёмки, заказав этот чудо-аппарат. Кто знает, может с Das Auge IV начнётся ваш путь к наградам всех престижных кинофестивалей.
        </span>
        <a
          class="btn"
          href="/product/1"
        >
          Подробнее
        </a>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
